### PR TITLE
[embedded-elt] Update Sling docs and examples

### PR DESCRIPTION
**Pull request by PedramNavid**
## Summary & Motivation

Updates Sling docs to use new API, deprecating the old one.
I moved the Relevant API section to the bottom as well, for an easier flow.

### CODE REVIEW

**Comment by PedramNavid** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20028?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* **#20029** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20029?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#20028** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20028?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#19998** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19998?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#19999** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19999?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/20028?utm_source=stack-comment).

File: docs/content/integrations/embedded-elt.mdx [hunk 1]
```diff
@@ -5,167 +5,253 @@ description: Lightweight ELT framework for building ELT pipelines with Dagster,
...
-3. Finally, create an asset that syncs between two connections. You can use the <PyObject module="dagster-embedded-elt.sling" object="build_sling_asset" /> factory for most use cases.
+Each step is explained in detail below:
 
 ---
 
-## Step 1: Setting up a Sling resource
+## Step 1: Setting up a Sling replication configuration
 
-A Sling resource is a Dagster resource that contains references to both a source connection and a target connection. Sling is versatile in what a source or destination can represent. You can provide arbitrary keywords to the <PyObject module="dagster-embedded-elt.sling" object="SlingSourceConnection" /> and <PyObject module="dagster-embedded-elt.sling" object="SlingTargetConnection" /> classes.
+Dagster's Sling integration is built around Sling's replication configuration. You may provide either a path to an existing `replication.yaml` file, or construct a dictionary that represents the configuration in Python.
```
Comment (cmpadden) [inline docs/content/integrations/embedded-elt.mdx:34]: Should we mention here that we recommend keeping this in version control alongside the Dagster project, or is that too limiting? For example, might there be some people who keep it in S3, or some artifact repository?

Comment (PedramNavid) [reply docs/content/integrations/embedded-elt.mdx:34]: I'll add something around how it needs to be colocated with your dagster code, or you need to read it in somehow.

File: docs/content/integrations/embedded-elt.mdx [hunk 2]
```diff
@@ -5,167 +5,253 @@ description: Lightweight ELT framework for building ELT pipelines with Dagster,
...
-```python
+The `name` parameter in the <PyObject module="dagster_embedded_elt.sling" object="SlingConnectionResource" /> should match the `SOURCE` and `TARGET` keys in the replication configuration.
 
-sling_resource = SlingResource(
-    source_connection=SlingSourceConnection(type="file"),
-    target_connection=SlingTargetConnection(
-        type="sqlite", connection_string="sqlite://path/to/sqlite.db"
-    ),
-)
+You may pass a connection string or arbitrary keyword arguments to the <PyObject module="dagster_embedded_elt.sling" object="SlingConnectionResource" /> to specify the connection details. See the <a href="https://docs.slingdata.io/connections/database-connections">Sling connections reference</a> for the specific connection types and parameters.
```
[the remark this answers is not in the document]
Comment (PedramNavid) [reply docs/content/integrations/embedded-elt.mdx:78]: good catch!

**Review by cmpadden — APPROVED** [review]

File: docs/content/integrations/embedded-elt.mdx [hunk 3]
```diff
@@ -5,167 +5,253 @@ description: Lightweight ELT framework for building ELT pipelines with Dagster,
...
 
-The simplest connection is a file connection, which can be defined as:
+Here's an example of a `replication.yaml` file:
 
-```python
-from dagster_embedded_elt.sling import SlingSourceConnection
-source = SlingSourceConnection(type="file")
-sling = SlingResource(source_connection=source, ...)
-```
+```yaml
```
Comment (erinkcochran87) [inline docs/content/integrations/embedded-elt.mdx:40]: It looks like step 2 includes three databases: Postgres, DuckDB, and Snowflake. Should we have both of these examples include all three so it's a 1:1 for each approach?

File: docs/content/integrations/embedded-elt.mdx [hunk 4]
```diff
@@ -5,167 +5,253 @@ description: Lightweight ELT framework for building ELT pipelines with Dagster,
...
+            type="snowflake",
+            host=EnvVar("SNOWFLAKE_HOST"),
+            user=EnvVar("SNOWFLAKE_USER"),
+            role="REPORTING",
+        ),
+    ]
 )
+```
+
+## Step 3: Define the Sling assets
```
Comment (erinkcochran87) [inline docs/content/integrations/embedded-elt.mdx:114]:
```suggestion
---

## Step 3: Define the Sling assets
```

[2]

File: docs/content/integrations/embedded-elt.mdx [hunk 5]
```diff
@@ -5,167 +5,253 @@ description: Lightweight ELT framework for building ELT pipelines with Dagster,
...
 
-1. First, create a <PyObject module="dagster-embedded-elt.sling" object="SlingResource" /> which is a container for the source and the target.
-2. In the <PyObject module="dagster-embedded-elt.sling" object="SlingResource" /> define both a <PyObject module="dagster-embedded-elt.sling" object="SlingSourceConnection" /> and a <PyObject module="dagster-embedded-elt.sling" object="SlingTargetConnection" /> which holds the source and target credentials that Sling will use to sync data.
-3. Finally, create an asset that syncs between two connections. You can use the <PyObject module="dagster-embedded-elt.sling" object="build_sling_asset" /> factory for most use cases.
+Each step is explained in detail below:
 
 ---
 
-## Step 1: Setting up a Sling resource
+## Step 1: Setting up a Sling replication configuration
```
Comment (erinkcochran87) [inline docs/content/integrations/embedded-elt.mdx:32]:
```suggestion
## Step 1: Set up a Sling replication configuration
```

[1] Let's drop the gerund in these headings

File: docs/content/integrations/embedded-elt.mdx [hunk 6]
```diff
@@ -5,167 +5,253 @@ description: Lightweight ELT framework for building ELT pipelines with Dagster,
...
+            user=EnvVar("SNOWFLAKE_USER"),
+            role="REPORTING",
+        ),
+    ]
 )
+```
+
+## Step 3: Define the Sling assets
+
+Now you can define a Sling asset using the <PyObject module="dagster_embedded_elt.sling" object="sling_assets" /> decorator. Dagster will read the replication configuration to produce Assets.
```
Comment (erinkcochran87) [inline docs/content/integrations/embedded-elt.mdx:116]:
```suggestion
Now you can define a Sling asset using the <PyObject module="dagster_embedded_elt.sling" object="sling_assets" decorator /> decorator. Dagster will read the replication configuration to produce assets.
```

File: docs/content/integrations/embedded-elt.mdx [hunk 7]
```diff
@@ -5,167 +5,253 @@ description: Lightweight ELT framework for building ELT pipelines with Dagster,
...
+        my_assets,
+    ],
+    resources={
+        "sling": sling_resource,
+    },
+)
 ```
 
----
+That's it! You should now be able to view your assets in Dagster and run the replication job.
```
Comment (erinkcochran87) [inline docs/content/integrations/embedded-elt.mdx:154]:
```suggestion
That's it! You should now be able to view your assets in the Dagster UI and run the replication job.
```

File: docs/content/integrations/embedded-elt.mdx [hunk 8]
```diff
@@ -5,167 +5,253 @@ description: Lightweight ELT framework for building ELT pipelines with Dagster,
...
 
 ## Examples
```
Comment (erinkcochran87) [inline docs/content/integrations/embedded-elt.mdx:156]:
```suggestion
---

## Examples
```

[2]

File: docs/content/integrations/embedded-elt.mdx [hunk 9]
```diff
@@ -5,167 +5,253 @@ description: Lightweight ELT framework for building ELT pipelines with Dagster,
...
 
 This is an example of how to setup a Sling sync between Postgres and Snowflake:
```
Comment (erinkcochran87) [inline docs/content/integrations/embedded-elt.mdx:158]: Thoughts on adding subheadings to these examples to make it easier to tell what they are?

File: docs/content/integrations/embedded-elt.mdx [hunk 10]
```diff
@@ -5,167 +5,253 @@ description: Lightweight ELT framework for building ELT pipelines with Dagster,
...
+        replication_config=replication_config,
+        dagster_sling_translator=DagsterSlingTranslator(),
+    )
 ```
+
+## Relevant APIs
+
+| Name                                                                              | Description                                                                    |
+| --------------------------------------------------------------------------------- | ------------------------------------------------------------------------------ |
+| <PyObject module="dagster_embedded_elt.sling" object="sling_assets" />            | The core Sling asset factory for building syncs                                |
```
Comment (erinkcochran87) [inline docs/content/integrations/embedded-elt.mdx:254]:
```suggestion
| <PyObject module="dagster_embedded_elt.sling" object="sling_assets" decorator />            | The core Sling asset factory for building syncs                                |
```

File: docs/content/integrations/embedded-elt.mdx [hunk 11]
```diff
@@ -5,167 +5,253 @@ description: Lightweight ELT framework for building ELT pipelines with Dagster,
...
+| <PyObject module="dagster_embedded_elt.sling" object="SlingResource" />           | The Sling Resource used for handing credentials to databases and object stores |
+| <PyObject module="dagster_embedded_elt.sling" object="DagsterSlingTranslator" />  | A translator for specifying how to map between Sling and Dagster types         |
+| <PyObject module="dagster_embedded_elt.sling" object="SlingConnectionResource" /> | A Sling connection resource for specifying the connection details              |
```
Comment (erinkcochran87) [inline docs/content/integrations/embedded-elt.mdx:257]: > the connection details

For what? I suggest including `resource` in here somewhere to make it more specific

File: docs/content/integrations/embedded-elt.mdx (hunk 5 above)
Comment (PedramNavid) [reply docs/content/integrations/embedded-elt.mdx:32]: drop the gerund is my favorite dubstep album

File: docs/content/integrations/embedded-elt.mdx (hunk 11 above)
Comment (PedramNavid) [reply docs/content/integrations/embedded-elt.mdx:265]: updated to: "A Sling connection resource for specifying database and storage connection credentials" how's that sound?

File: docs/content/integrations/embedded-elt.mdx (hunk 9 above)
Comment (PedramNavid) [reply docs/content/integrations/embedded-elt.mdx:164]: ooh i like it! picasso

File: docs/content/integrations/embedded-elt.mdx (hunk 3 above)
Comment (PedramNavid) [reply docs/content/integrations/embedded-elt.mdx:40]: You can only have two in the a single replication config, so I'll remove one from step 2. I don't think it's necessary.

File: docs/content/integrations/embedded-elt.mdx [hunk 12]
```diff
@@ -5,167 +5,253 @@ description: Lightweight ELT framework for building ELT pipelines with Dagster,
...
 
-## Getting started
+1. First, define a <a href="https://docs.slingdata.io/sling-cli/run/configuration/replication"> `replication.yaml`</a> file that specifies the source and target connections, as well as which streams to sync from.
 
-To get started with `dagster-embedded-elt` and Sling, familiarize yourself with <a href="https://docs.slingdata.io/sling-cli/run">Sling</a> by reading their docs which describe how sources and targets are configured.
+2. Next, create a <PyObject module="dagster_embedded_elt.sling" object="SlingResource" /> and pass a list of <PyObject module="dagster_embedded_elt.sling" object="SlingConnectionResource" /> for each connection to the `connection` parameter, ensuring you name the resource using the same name given to the connection in the Sling configuration.
 
-The typical pattern for building an ELT pipeline with Sling has three steps:
+3. Use the <PyObject module="dagster_embedded_elt.sling" object="sling_assets" /> decorator to define an asset that will run the Sling replication job and yield from the <PyObject module="dagster_embedded_elt.sling" object="SlingResource" method="replicate" /> method to run the sync.
```
[the remark this answers is not in the document]
Comment (PedramNavid) [reply docs/content/integrations/embedded-elt.mdx:26]: yes!

**Pull request merged**